flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/DefaultSchemaResolver.java

```diff
@@ -213,11 +213,10 @@ private void validateTimeColumn(String columnName, List<Column> columns) {
                             columns.stream().map(Column::getName).collect(Collectors.toList())));
         }
         final LogicalType timeFieldType = timeColumn.get().getDataType().getLogicalType();
-        if (!hasRoot(timeFieldType, LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE)
-                || getPrecision(timeFieldType) != 3) {
+        if (!supportedWatermarkType(timeFieldType) || getPrecision(timeFieldType) != 3) {
```

Comment:
`supportedWatermarkType` looks weird because time field of cause can't be watermark. I think you may want to use `canBeTimeAttributeType` here?